main.go

```diff
@@ -16,13 +16,139 @@
+	if err != nil {
+		http.Error(w, "error initializing Rekor client", http.StatusInternalServerError)
+		return
+	}
+	uuids, err := cosign.FindTLogEntriesByPayload(ctx, rekorClient, reqBody)
+	if err != nil || len(uuids) == 0 {
+		http.Error(w, "error fetching tlog entries", http.StatusInternalServerError)
+		return
+	}
+	uuid := uuids[len(uuids)-1]
+
+	// Verify tlog entry to make sure it is actually in the log.
+	entry, err := verifyTLogEntry(ctx, rekorClient, uuid)
+	if err != nil {
+		http.Error(w, "error verifying tlog entry", http.StatusInternalServerError)
+		return
+	}
+
+	// Extract certificate and get repo reference & path.
+	certs, err := extractCerts(entry)
+	if err != nil || len(certs) == 0 {
+		http.Error(w, "error extracting certificate from entry", http.StatusInternalServerError)
+		return
+	}
+	cert := certs[0]
+	var repoRef string
+	var repoPath string
+	for _, ext := range cert.Extensions {
+		if ext.Id.String() == "1.3.6.1.4.1.57264.1.6" {
+			repoRef = string(ext.Value)
+		}
+		if ext.Id.String() == "1.3.6.1.4.1.57264.1.5" {
+			repoPath = string(ext.Value)
+		}
+	}
+
+	if len(repoRef) == 0 || len(repoPath) == 0 {
+		http.Error(w, "error extracting repo ref or path from certificate", http.StatusInternalServerError)
+		return
+	}
+
+	// Split repo path into owner and repo name.
+	ownerName := repoPath[:strings.Index(repoPath, "/")]
+	repoName := repoPath[strings.Index(repoPath, "/")+1:]
+
+	// Get workflow file from repo reference
+	client := github.NewClient(nil)
+	opts := &github.RepositoryContentGetOptions{Ref: repoRef}
+	contents, _, _, err := client.Repositories.GetContents(ctx, ownerName, repoName, ".github/workflows/scorecards.yml", opts)
+	if err != nil {
+		http.Error(w, "error downloading workflow contents from repo", http.StatusInternalServerError)
+		return
+	}
+
+	workflowContent, err := contents.GetContent()
+	if err != nil {
+		http.Error(w, "error decoding workflow contents", http.StatusInternalServerError)
+		return
+	}
+	// ioutil.WriteFile("test.yml", []byte(workflowContent), 0644)
+
+	// Next steps: verify workflow contents using actionlint.
+	// Currently just checks if there is a job titled "Scorecards analysis", but this will
+	// later be expanded to do more comprehensive scanning.
+	workflow, lintErrs := actionlint.Parse([]byte(workflowContent))
+	if lintErrs != nil {
+		http.Error(w, "error parsing workflow contents", http.StatusInternalServerError)
+		return
+	}
+
+	jobs := workflow.Jobs
+	analysisJob := jobs["analysis"]
+	if analysisJob == nil {
+		http.Error(w, "analysis job not present in workflow", http.StatusNotFound)
+		return
+	}
+
+	analysisJobName := analysisJob.Name.Value
+	if analysisJobName != "Scorecards analysis" {
+		http.Error(w, "scorecard analysis job not found", http.StatusNotFound)
+		return
+	}
+
+	// Return result.
+}
+
 func main() {
 	http.HandleFunc("/", httpHandler)
+	http.HandleFunc("/verifyProject", verifySignature)
```

Comment:
Update `httpHandler` function to call `verifySignature` for `http.MethodPost` instead.